src/Database/MysqlDatabase.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Friendica\Database;
+
+use Friendica\Core\Config\Cache\IConfigCache;
+use Friendica\Util\Profiler;
+
+class MysqlDatabase implements IDatabase, IDatabaseLock
+{
+	private $connected;
+
+	public function __construct(IConfigCache $configCache, Profiler $profiler, $serveraddr, $user, $pass, $db, $charset = null)
+	{
+		$this->connected = DBA::connect($configCache, $profiler, $serveraddr, $user, $pass, $db, $charset);
+	}
+
+	function isConnected()
+	{
+		return $this->connected;
+	}
+
+	function disconnect()
+	{
+		DBA::disconnect();
+	}
+
+	function reconnect()
+	{
+		DBA::reconnect();
+		$this->connected = DBA::connected();
+	}
+
+	function getConnection()
+	{
+		return DBA::getConnection();
+	}
+
+	function serverInfo()
+	{
+		return DBA::serverInfo();
+	}
+
+	function databaseName()
+	{
+		return DBA::databaseName();
+	}
+
+	function exists($table, array $condition)
+	{
+		return DBA::exists($table, $condition);
+	}
+
+	function count($table, array $condition = [])
+	{
+		return DBA::count($table, $condition);
+	}
+
+	function fetch($stmt)
+	{
+		return DBA::fetch($stmt);
+	}
+
+	function transaction()
+	{
+		return DBA::transaction();
+	}
+
+	function commit()
+	{
+		return DBA::commit();
+	}
+
+	function rollback()
+	{
+		return DBA::rollback();
+	}
+
+	function insert($table, array $param, $on_duplicate_update = false)
+	{
+		return DBA::insert($table, $param, $on_duplicate_update);
+	}
+
+	function delete($table, array $conditions, $cascade = true)
+	{
+		return DBA::delete($table, $conditions, [$cascade]);
+	}
+
+	function update($table, array $fields, array $condition, array $old_fields = [])
+	{
+		return DBA::delete($table, $fields, $condition, $old_fields);
+	}
+
+	function select($table, array $fields = [], array $condition = [], array $params = [])
+	{
+		return DBA::select($table, $fields, $condition, $params);
```

Comment:
This isn't the right wrapping direction. In the same spirit we transitioned from `q()` to `DBA::p()` by making `q()` a shell making a call to `DBA::p()`, these methods should be wrapping the other way, meaning that `DBA::select` should retrieve the DB object and call `MysqlDatabase->select` which should have the actual logic.

Reply:
BTW: Concerning MySQL and PostgreSQL I have to see if we really need different classes for this. Both are using PDO and many DBA classes don't do direct MySQL calls but calling other DBA functions. So we most likely only have to separate some of these class calls at all.

Reply:
Hmm ... that's what I would solve with abstraction:
```php
abstract class DBA
{
     public function otherDbaFunction()
     {
          return something;
     }
}

class MysqlDba extends DBA
{
   public function specificMysqlFunction()
   {
        return $this->otherDbaFunction();
   }
}

class PostgreSqlDba extends DBA
{
   public function specificPsqlFunction()
   {
        return $this->otherDbaFunction();
   }
}
```